---
title: "Metagenomics: Text EDA"
output: html_notebook
---

```{r include=FALSE}
library(tidyverse)
```

# Load Gtdb-tk, select and separate `closest_placement_taxonomy` column and write to csv
```{r eval=FALSE, warning=FALSE, include=FALSE}
read_tsv('./Meta_Gtdbtk/24_gtdbtk.bac120.summary.tsv') %>% 
  select(closest_placement_taxonomy) %>% 
  separate(col = 1,
           into = c("domain", "phylum", "class", "order", "family", "genus", "species"), 
           sep = ";") %>%  
  # select(species) %>% 
  #write_excel_csv('~/Desktop/24_gtdbtk_taxid.csv')

read_tsv('./Meta_Gtdbtk/25_gtdbtk.bac120.summary.tsv') %>% 
  select(closest_placement_taxonomy) %>% 
  separate(col = 1,
           into = c("domain", "phylum", "class", "order", "family", "genus", "species"), 
           sep = ";") %>% 
  # select(species) %>% 
  #write_excel_csv('~/Desktop/25_gtdbtk_taxid.csv')

read_tsv('./Meta_Gtdbtk/38_gtdbtk.bac120.summary.tsv') %>% 
  select(closest_placement_taxonomy) %>% 
  separate(col = 1,
           into = c("domain", "phylum", "class", "order", "family", "genus", "species"), 
           sep = ";") %>% 
  # select(species) %>% 
  #write_excel_csv('~/Desktop/38_gtdbtk_taxid.csv')
```

# Load Gtdb-tk output
```{r message=FALSE, warning=FALSE}
gtdb_24 <- read_tsv('./Meta_Gtdbtk/24_gtdbtk.bac120.summary.tsv') 

gtdb_25 <- read_tsv('./Meta_Gtdbtk/25_gtdbtk.bac120.summary.tsv')

gtdb_38 <- read_tsv('./Meta_Gtdbtk/38_gtdbtk.bac120.summary.tsv') 
```

# Load checkM2 quality report 
```{r message=FALSE, warning=FALSE}
checkm_24 <- read_tsv('./checkM2_original_AND_edit_quality_report/sample24_quality_report.tsv')

checkm_25 <- read_tsv('./checkM2_original_AND_edit_quality_report/sample25_quality_report.tsv')

checkm_38 <- read_tsv('./checkM2_original_AND_edit_quality_report/sample38_quality_report.tsv')
```

# Load EDITED checkM2 results 
```{r}
checkm_edit_24 <- read_csv('./checkM2_original_AND_edit_quality_report/sample24_EDITED_quality_report.csv')

checkm_edit_25 <- read_csv('./checkM2_original_AND_edit_quality_report/sample25_EDITED_quality_report.csv')

checkm_edit_38 <- read_csv('./checkM2_original_AND_edit_quality_report/sample38_EDITED_quality_report.csv')
```

# Plot EDITED checkM2 results
```{r}
# Merge dataframe after samples
plot_df_24 <- checkm_edit_24 %>% mutate(Sample = "Sample 24") %>% as_tibble()
plot_df_25 <- checkm_edit_25 %>% mutate(Sample = "Sample 25") %>% as_tibble()
plot_df_38 <- checkm_edit_38 %>% mutate(Sample = "Sample 38") %>% as_tibble()

merged_plot <- plot_df_24 %>% full_join(plot_df_25) %>% full_join(plot_df_38)

# Plot
merged_plot %>%  ggplot(aes(x=completeness,y=contamination,colour=Sample))+geom_point()+ggtitle("Bins after checkM2")

merged_plot  %>% filter(completeness>75) %>% filter(contamination<25) %>%  ggplot(aes(x=completeness,y=contamination,colour=Sample))+geom_point()+ggtitle("Our dRep settings")

merged_plot %>% filter(completeness>90 & contamination<5) %>% ggplot(aes(x=completeness,y=contamination,colour=Sample))+geom_point()+ggtitle("Only HQ bins")
```

# contaminate/completeness plot with plots 

Our settings: complete=75 and contamination=25
Medium quality: complete=50-90 and contamination=0-10
High quality: complete=90-100 and contamination=0-5
```{r}
# All genome bins (dRep highlighted)
merged_plot %>% ggplot(aes(x=completeness,y=contamination,colour=Sample)) + 
  scale_color_manual(values = c("Sample 24" = "blue", "Sample 25" = "red", "Sample 38" = "black")) +
  geom_point() +
  geom_rect(aes(xmin = 75, xmax = Inf, ymin = -Inf, ymax = 25), alpha = 0.005, fill = "white") + # dRep settings
  theme_bw() +
  guides(colour=guide_legend(override.aes = list(linetype = c(0,0,0)))) + # Remove boxes around legend
  # ggtitle("All genome bins after checkM2") +
  theme(legend.position="bottom")

#ggsave('~/Desktop/bin_all.png')

# dRep bins (MQ and HQ bins highlighted)
merged_plot %>% 
  filter(completeness>75 & contamination<25) %>% 
  ggplot(aes(x=completeness,y=contamination,colour=Sample)) + 
  scale_color_manual(values = c("Sample 24" = "blue", "Sample 25" = "red", "Sample 38" = "black")) +
  scale_x_continuous(limits=c(75,100)) + 
  scale_y_continuous(limits=c(0,25)) +
  geom_point() +
  geom_rect(aes(xmin = -Inf, xmax = Inf, ymin = -Inf, ymax = 10), alpha = 0.005, fill = "green") + #Medium quality bin
  geom_rect(aes(xmin = 95, xmax = Inf, ymin = -Inf, ymax = 5), alpha = 0.01, fill = "green") + # High quality bin
  guides(colour=guide_legend(override.aes = list(linetype = c(0,0,0)))) + # Remove boxes around legend 
  theme_bw() +
  ggtitle('Bins used in dRep', subtitle = 'Lightgreen=MQ bins Darkgreen=HQ bins')
```

# Zoomed graph without title and subtitle
```{r}
# dRep bins (MQ and HQ bins highlighted)

merged_plot %>% 
  filter(completeness>75 & contamination<25) %>% 
  ggplot(aes(x=completeness,y=contamination,colour=Sample)) + 
  scale_color_manual(values = c("Sample 24" = "blue", "Sample 25" = "red", "Sample 38" = "black")) +
  scale_x_continuous(limits=c(75,100)) + 
  scale_y_continuous(limits=c(0,25)) +
  geom_point() +
  geom_rect(aes(xmin = -Inf, xmax = Inf, ymin = -Inf, ymax = 10), alpha = 0.005, fill = "green") + #Medium quality bin
  # geom_rect(aes(xmin = 95, xmax = Inf, ymin = -Inf, ymax = 5), alpha = 0.01, fill = "green") + # High quality bin
  guides(colour=guide_legend(override.aes = list(linetype = c(0,0,0)))) + # Remove boxes around legend 
  theme_bw() +
  guides(color = FALSE, size = FALSE)

#ggsave('~/Desktop/bin_mediumquality.png')
```

# Example of rectangle plot
```{r eval=FALSE, message=FALSE, warning=FALSE, include=FALSE}
y_lim <- 20 # a custom threshold for the horizontal line

ggplot() +
  geom_point(data = mtcars,
             aes(x = hp, y = mpg)) +
  geom_hline(yintercept = y_lim) +
  # Shade area under y_lim
  geom_rect(aes(xmin = -Inf, xmax = Inf, ymin = -Inf, ymax = y_lim),
            alpha = 1/5,
            fill = "blue") +
  # Shade area above y_lim
  geom_rect(aes(xmin = -Inf, xmax = Inf, ymin = y_lim, ymax = Inf),
            alpha = 1/5,
            fill = "red")
```


# Example of multiple rectangle plots
```{r eval=FALSE, message=FALSE, warning=FALSE, include=FALSE}
d=data.frame(x1=c(1,3,1,5,4), x2=c(2,4,3,6,6), y1=c(1,1,4,1,3), y2=c(2,2,5,3,5), t=c('a','a','a','b','b'), r=c(1,2,3,4,5))
ggplot() + 
scale_x_continuous(name="x") + 
scale_y_continuous(name="y") +
geom_rect(data=d, mapping=aes(xmin=x1, xmax=x2, ymin=y1, ymax=y2, fill=t), color="black", alpha=0.5) +
geom_text(data=d, aes(x=x1+(x2-x1)/2, y=y1+(y2-y1)/2, label=r), size=4)
```

# Number of bins for dRep
```{r}
# checkm_edit_24 %>% filter(completeness>75 & contamination<25) %>% count() 
# checkm_edit_25 %>% filter(completeness>75 & contamination<25) %>% count() 
# checkm_edit_38 %>% filter(completeness>75 & contamination<25) %>% count() 

checkm_edit_24 %>% count()
checkm_edit_25 %>% count()
checkm_edit_38 %>% count()
```

# Coverage table (Metabat2/JGI)
```{r}
coverage_24 <- read_tsv('./jgi_coverage_table/coverage_ERR2597524.txt')

coverage_25 <- read_tsv('./jgi_coverage_table/coverage_ERR2597525.txt')

coverage_38 <- read_tsv('./jgi_coverage_table/coverage_ERR2597538.txt')

```

# ALR - re-ordering them
```{r}
alr_24 <- read_csv('./ALR_24.csv') %>% mutate(Sample = "Sample 24") 
alr_25 <- read_csv('./ALR_25.csv') %>% mutate(Sample = "Sample 25")
alr_38 <- read_csv('./ALR_38.csv') %>% mutate(Sample = "Sample 38")

alr_composite <- alr_24 %>% full_join(alr_25) %>% full_join(alr_38) %>% select(c("Sample", "Class", "logFPKM"))

alr_composite %>%
  ggplot(aes(fill=Sample, x=reorder(Class,-logFPKM), y=logFPKM)) +
  geom_bar(position="dodge", stat="identity") + 
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.5, hjust=0.5)) +
  theme(legend.position="bottom") +
  xlab(label="AMR gene class")
  
#ggsave('~/Desktop/ALR_new.png')
```


